src/OpenTelemetry/Metrics/MeterSdk.cs

```diff
@@ -115,11 +127,21 @@ public void Collect()
                     }
                 }
 
-                foreach (var longObserver in this.longObservers)
+                foreach (var doubleObserver in this.doubleObservers)
                 {
-                    var metricName = longObserver.Key;
-                    var measureInstrument = longObserver.Value;
-                    foreach (var handle in measureInstrument.GetAllHandles())
+                    var metricName = doubleObserver.Key;
+                    var observerInstrument = doubleObserver.Value;
+                    try
+                    {
+                        // TODO: Decide if we want to enforce a timeout.
```

Comment:
Can you raise an issue to track this `TODO` please.

Reply:
done